pkg/buildplan/hydrate.go

```diff
@@ -212,7 +214,7 @@ func (b *BuildPlan) sanityCheck() error {
 	// Ensure all artifacts have an associated ingredient
 	// If this fails either the API is bugged or the hydrate logic is bugged
 	for _, a := range b.Artifacts() {
-		if len(a.Ingredients) == 0 {
+		if raw.IsStateToolMimeType(a.MimeType) && len(a.Ingredients) == 0 {
```

Comment:
We're only concerned with state tool artifacts having ingredients now?

Reply:
They're the only type of artifact that is guaranteed to have an ingredient, so the only thing we can reasonably validate.